chains/chains.go

```diff
@@ -121,6 +121,53 @@ func Predict(ctx context.Context, c Chain, inputValues map[string]any, options .
 	return outputValue, nil
 }
 
+const _defaultApplyMaxNumberWorkers = 5
+
+// Apply executes the chain for each of the inputs asynchronously.
+func Apply(ctx context.Context, c Chain, inputValues []map[string]any, maxWorkers int, options ...ChainCallOption) ([]map[string]any, error) { // nolint:lll
+	if maxWorkers <= 0 {
+		maxWorkers = _defaultApplyMaxNumberWorkers
+	}
+
+	inputJobs := make(chan map[string]any, len(inputValues))
+	resultsChan := make(chan struct {
+		result map[string]any
+		err    error
+	}, len(inputValues))
+	defer close(inputJobs)
+	defer close(resultsChan)
+
+	for w := 0; w < maxWorkers; w++ {
```

Comment:
can you double check this doesn't suffer from the go gotcha https://github.com/golang/go/issues/60078

Reply:
I dont think it does. The variable w is not used in the anonymous function.